source/hardware/controllers/pioneer_ddj_400.rst

```diff
@@ -0,0 +1,185 @@
+Pioneer DDJ-400
+===============
+
+.. sectionauthor::
+   jusko <justin.kourie@gmail.com>
+
+.. figure:: ../../_static/controllers/pioneer_ddj_400.svg
+   :align: center
+   :width: 100%
+   :figwidth: 100%
+   :alt: Pioneer DDJ-400 (schematic view)
+   :figclass: pretty-figures
+
+   Pioneer DDJ-400 (schematic view)
+
+The Pioneer DDJ-400 is a 2 deck USB controller with integrated audio interface designed for rekordbox.
+
+- `Manufacturer's Product Page <https://www.pioneerdj.com/en-us/product/controller/ddj-400/black/overview/>`__
+- `Manufacturer's User Manual <http://docs.pioneerdj.com/Manuals/DDJ_400_DRI1551A_manual/>`__
+- `Manufacturer's Firmware <https://www.pioneerdj.com/en/support/software/controller/ddj-400/>`__
+- `Midi Mappings <https://www.pioneerdj.com/-/media/pioneerdj/software-info/controller/ddj-400/ddj-400_midi_message_list_e1.pdf>`__
+- `Hardware Diagram <https://www.pioneerdj.com/-/media/pioneerdj/software-info/controller/ddj-400/ddj-400_hardwarediagram_rekordboxdj_e1.pdf?la=en-us>`__
+- `Mapping Forum Thread <https://mixxx.org/forums/viewtopic.php?f=7&t=12113>`__
+
+.. versionadded:: 2.3.0
+
+Drivers
+-------
+
+You can download the latest firmware from
```

Comment:
This is a USB audio and MIDI class compliant device, right? So it works fully with Linux other than the firmware updater, right? This should be documented explicitly.

It seems Pioneer offers no drivers for any OS including Windows. So I guess Windows users should use the WDM-KS sound API because there is no ASIO driver?

Reply:
>This is a USB audio and MIDI class compliant device, right? So it works fully with Linux other than the firmware updater, right? This should be documented explicitly.

Roger that :+1: 

> It seems Pioneer offers no drivers for any OS including Windows. So I guess Windows users should use the WDM-KS sound API because there is no ASIO driver?

I'll change the section to just discuss the firmware as the bug was a show stopper. Don't know much about windows or mac drivers, so not much I can add in that respect.